### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by andrzeng**
Working version of eigengame without documentation

### CODE REVIEW

File: NeuralABC_tools/eigengame.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import numpy as np
+
+def calc_penalties(data, vectors, index):
+    """
+    
+    """
```
Comment (zklsmr) [inline NeuralABC_tools/eigengame.py]: Please add docstring

File: NeuralABC_tools/eigengame.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        penalties += (np.dot(np.dot(data, vec), result) /
+         np.dot(result, result)
+        ) * result
+
+    return penalties 
+     
+def eigengame(data, n, epochs=100, learning_rate=0.1):
+    """
+        
+    """
```
Comment (zklsmr) [inline NeuralABC_tools/eigengame.py]: Please add docstring

**Comment by zklsmr** [comment]
Looks great! please document the functions and include a call for the script in ```__init__.py``` (check that file for examples).

File: NeuralABC_tools/eigengame.py [hunk 3]
```diff
@@ -39,13 +37,10 @@ def eigengame(data, n_components, epochs=100, learning_rate=0.1):
...
     Parameters:
     ----------------
-    data: a Numpy array containing the data to run PCA on
-    
-    n_components: the number of principal components to extract
-    
-    epochs: the number of iterations to calculate each eigenvector. Default = 100
-    
-    learning_rate: Learning rate of the algorithm. Default = 0.1
+    :param data: (array), required: a Numpy array containing the data to run PCA on
```
Comment (steelec) [inline NeuralABC_tools/eigengame.py]: can you specify the form of this array please? i.e., it is a 2d array with what expectations for each dimension?

**Review by steelec — CHANGES_REQUESTED** [review]

File: NeuralABC_tools/eigengame.py (hunk 3 above)
Comment (andrzeng) [reply NeuralABC_tools/eigengame.py]: I made the description more specific

Comment (steelec) [reply NeuralABC_tools/eigengame.py]: good, which axis is which for the PCA though

e.g., from sklearn PCA:
```
Parameters
Xarray-like of shape (n_samples, n_features)
    Training data, where n_samples is the number of samples and n_features is the number of features.
```

**Review by steelec — APPROVED** [review]

**Comment by steelec** [comment]
merging, we can change the init file to hide the internal calculations in the future as necessary
